File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    
    data.sort(key=lambda x: (x[col-1], -x[0]))
    
    answer = sum(j % row_begin for j in data[row_begin - 1])
    
    for i in range(row_begin, row_end):
        answer ^= sum(j % (i+1) for j in data[i])
    return answer
        
    

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        Arrays.sort(data, Comparator.comparingInt((int[] d) -> d[col - 1]).thenComparingInt((d) -> -d[0]));
        
        int result = 0;
        for (int i=row_begin - 1; i < row_end; i++) {
            int[] row = data[i];
            int index = i + 1;
            result ^= Arrays.stream(row).map(d -> d % index).sum();
        }
        
        return result;
    }
}